commit 9257845512595edd65f7eb368abb557d0a6d90f2
Author: Mark Thomas <markt@apache.org>
Date:   Wed Mar 11 20:01:23 2015 +0000

    Restore volatile for closed. Not all reads are inside a sync.
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1665988 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/tomcat/util/net/AprEndpoint.java b/java/org/apache/tomcat/util/net/AprEndpoint.java
index e2dee48..9245358 100644
--- a/java/org/apache/tomcat/util/net/AprEndpoint.java
+++ b/java/org/apache/tomcat/util/net/AprEndpoint.java
@@ -2351,7 +2351,7 @@ public class AprEndpoint extends AbstractEndpoint<Long> {
         private final ByteBuffer sslOutputBuffer;
 
         private final Object closedLock = new Object();
-        private boolean closed = false;
+        private volatile boolean closed = false;
 
         // This field should only be used by Poller#run()
         private int pollerFlags = 0;
